### PR TITLE
Introduce a production-ready image for Portus

### PR DESCRIPTION
**Pull request by flavio**
This commit introduces the Dockerfile and all the resources needed to build a production-ready image for Portus.

I'll add a kubernetes reference deployment under a new directory.

Once this PR is merged I'll create other branches with small changes to build the image out of the `2.0` and `2.0-git` packages.

### CODE REVIEW

**Comment by flavio** [comment]
@jordimassaguerpla please take a look at the PR, you might be interested by it too.

File: derived_images/portus/README.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    have only the runtime dependency; no build time dependency is ever installed.
+  * The default user for this image is the `wwwrun` one, which is created by the
+    Apache package.
+  * The root account is locked, that means it's not possible to switch user via
+    `su` or `sudo`.
+
+SUSE's containers team is constantly working to automate the release process
+of Portus and of this image to ensure it stays up-to-date and secure.
+
+When deploying this image make sure to add to add all the required keys and
```
Comment (mssola) [inline derived_images/portus/README.md]: Typo: "make sure to add to add all" -> "make sure to add all"

File: derived_images/portus/README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+soon release a reference implementation.
+
+### Environment variables
+
+The following environment variables must be defined to run the image:
+
+Security related settings:
+
+  * `PORTUS_SECRET_KEY_BASE`: you can generate it using `openssl rand -hex 64`
+  * `PORTUS_PORTUS_PORTUS_PASSWORD`: you can generate it using `openssl rand -hex 64`
```
Comment (mssola) [inline derived_images/portus/README.md]: `PORTUS_PORTUS_PORTUS_PASSWORD` what ? :stuck_out_tongue_closed_eyes:

File: derived_images/portus/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# That's why we use this trick.
+RUN ln -sf /dev/stdout /var/log/apache2/access_log
+RUN ln -sf /dev/stderr /var/log/apache2/error_log
+# The stable release of Portus is not logging to STDOUT by default when
+# used in production mode. We have to use this temporary workaround
+RUN ln -sf /dev/stdout /srv/Portus/log/production.log
+
+# portus configuration
+ADD database.yml /srv/Portus/config/
+ADD secrets.yml /srv/Portus/config/
```
Comment (mssola) [inline derived_images/portus/docker/Dockerfile]: `ADD database.yml secrets.yml /srv/Portus/config/`

File: derived_images/portus/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+ADD init /init
+ADD check_db.rb /check_db.rb
```
Comment (mssola) [inline derived_images/portus/docker/Dockerfile:45]: `ADD init check_db.rb /`

File: derived_images/portus/docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+EXPOSE 80
+EXPOSE 443
```
Comment (mssola) [inline derived_images/portus/docker/Dockerfile]: `EXPOSE 80 443`

**Comment by mssola** [comment]
LGTM :clap: :tada:

**Comment by flavio** [comment]
@mssola thanks for the review. I fixed all the issues you mentioned.

File: derived_images/portus/README.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    Apache package.
+  * The root account is locked, that means it's not possible to switch user via
+    `su` or `sudo`.
+
+SUSE's containers team is constantly working to automate the release process
+of Portus and of this image to ensure it stays up-to-date and secure.
+
+When deploying this image make sure to add all the required keys and
+certificates at runtime instead of adding them to a Docker image.
+
```
Comment (jordimassaguerpla) [inline derived_images/portus/README.md:27]: I like the security chapter :) :+1: +1

**Comment by jordimassaguerpla** [comment]
lgtm :+1:

**Pull request merged**